six/modules/c++/six.sicd/tests/test_compare_sicd_meshes.cpp

```diff
@@ -0,0 +1,352 @@
+                     const six::Mesh& testMesh)
+{
+    const types::RowCol<size_t> baselineDims = baselineMesh.getMeshDims();
+    const types::RowCol<size_t> testDims = testMesh.getMeshDims();
+    if (baselineDims != testDims)
+    {
+        std::cerr << "Failure - Mesh dimensions do not match: "
+                  << baselineSicdPathname << " had dimensions (row = "
+                  << baselineDims.row << ", col = " << baselineDims.col << "), "
+                  << testSicdPathname << " had dimensions (row = "
+                  << testDims.row << ", col = " << testDims.col << ")"
+                  << std::endl;
+
+        return false;
+    }
+
+    return true;
+}
+
+bool compareMeshData(const std::string& baselineSicdPathname,
+                     const std::vector<double>& baselineData,
+                     const std::string& testSicdPathname,
+                     const std::vector<double>& testData,
+                     const types::RowCol<size_t>& dims)
+{
+    if (baselineData.size() != testData.size())
+    {
+        std::cerr << "Failure - Mesh data sizes do not match: "
+                  << baselineSicdPathname << " size = " << baselineData.size()
+                  << ", " << testSicdPathname << " size = " << testData.size()
+                  << std::endl;
+
+        return false;
+    }
+
+    if (baselineData.size() != dims.area())
+    {
+        std::cerr << "Failure - Dimensions (row = " << dims.row << ", col = "
+                  << dims.col << ") do not match num elements in mesh ("
+                  << baselineData.size() << " elements)." << std::endl;
+
+        return false;
+    }
+
+    for (size_t row = 0; row < dims.row; ++row)
+    {
+        for (size_t col = 0; col < dims.col; ++col)
+        {
+            const size_t idx = row * dims.col + col;
+            if (baselineData[idx] != testData[idx])
+            {
+                std::cerr << "Failure - Values do not match at row = "
+                          << row << ", col = " << col << ": "
+                          << baselineSicdPathname << " value = "
+                          << baselineData[idx] << ", " << testSicdPathname
+                          << " value = " << testData[idx] << std::endl;
+
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+bool comparePlanarCoordinateMeshes(
+        const std::string& baselineSicdPathname,
+        const six::sicd::PlanarCoordinateMesh& baselineMesh,
+        const std::string& testSicdPathname,
+        const six::sicd::PlanarCoordinateMesh& testMesh)
+{
+    std::cout << "Comparing X coordinates..." << std::endl;
+    bool success = compareMeshData(
+            baselineSicdPathname,
+            baselineMesh.getX(),
+            testSicdPathname,
+            testMesh.getX(),
+            testMesh.getMeshDims());
+
+    std::cout << "Comparing Y coordinates..." << std::endl;
+    success &= compareMeshData(
```

Comment:
The compiler will do short-circuit evaluation so if success is false from an earlier step, it won't call the function.  So these all need to change to something like...
```
success = compareMeshData(...) && success;
```

Reply:
I don't think this will short-circuit due to &= being a bitwise operation.

Reply:
Oh!  You are right - ignore that comment.